### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by coinblockers**

### CODE REVIEW

File: lib/jobManager.js [hunk 1]
```diff
@@ -19,44 +19,21 @@ var ExtraNonceCounter = function (configInstanceId) {
     this.size = 4; //bytes
 };
 
+function randomInt(low, high) {
```
Comment (StarbuckBG) [inline lib/jobManager.js]: I have revised the changes and reverted them, as they are not actually doing any job at the moment

File: lib/jobManager.js [hunk 2]
```diff
@@ -182,38 +159,43 @@ var JobManager = module.exports = function JobManager(options) {
...
+        if (!job.registerSubmit(hdr,sol)) {
+            return shareError([22, 'duplicate share']);
+        }
 
         var headerBuffer = job.serializeHeader(nTime, nonce); // 144 bytes (doesn't contain soln)
         var headerSolnBuffer = new Buffer.concat([headerBuffer, new Buffer(soln, 'hex')]);
         var headerHash = util.sha256d(headerSolnBuffer);
         var headerBigNum = bignum.fromBuffer(headerHash, {endian: 'little', size: 32});
 
-        var blockHashInvalid;
```
Comment (StarbuckBG) [inline lib/jobManager.js]: Why this is moved to bottom ?

File: lib/jobManager.js [hunk 3]
```diff
@@ -222,12 +204,17 @@ var JobManager = module.exports = function JobManager(options) {
...
+        var blockHash;
+        var blockHex;
+        var blockHashInvalid;
+
         //check if block candidate
         if (headerBigNum.le(job.target)) {
             blockHex = job.serializeBlock(headerBuffer, new Buffer(soln, 'hex')).toString('hex');
             blockHash = util.reverseBuffer(headerHash).toString('hex');
-        }
-        else {
```
Comment (StarbuckBG) [inline lib/jobManager.js]: No need of this change

File: lib/jobManager.js [hunk 4]
```diff
@@ -237,11 +224,10 @@ var JobManager = module.exports = function JobManager(options) {
                 //Check if share matched a previous difficulty from before a vardiff retarget
                 if (previousDifficulty && shareDiff >= previousDifficulty) {
                     difficulty = previousDifficulty;
-                }
-                else {
```
Comment (StarbuckBG) [inline lib/jobManager.js]: No need of this change

File: lib/pool.js [hunk 5]
```diff
@@ -481,7 +480,8 @@ var pool = module.exports = function pool(options, authorizeFn) {
                 }
 
 
-                this.sendMiningJob(_this.jobManager.currentJob.getJobParams());
```
Comment (StarbuckBG) [inline lib/pool.js]: Is there a point of doing this ?

Comment (coinblockers) [reply lib/pool.js]: @StarbuckBG split is not necessary here.

File: lib/jobManager.js (hunk 2 above)
Comment (coinblockers) [reply lib/jobManager.js]: Code cleanups

**Pull request closed without merging**